### PR TITLE
amcrest-tui: Add Text User Interface to visualize snapshots

### PR DESCRIPTION
**Pull request by dougsland**
Users might want to not have X window so let's provide to them
a tool to visualize the snapshots by framebuffer.

### CODE REVIEW

**Comment by tchellomello** [comment]
@dougsland, This looks awesome!! Great idea!!  I'm adding this on my plate to test. 

In the meantime, what do you think to add a screenshot on the readme for our users?

**Comment by dougsland** [comment]
It's not about gui interface#12, this is a different issue that we will address. This PR is about TUI. :) 
I have updated the README as well, please take a look when you have free time.

**Comment by tchellomello** [comment]
@dougsland  yes makes sense. me--

File: README.md [hunk 1]
```diff
@@ -88,5 +88,26 @@ port: 80
 $ amcrest-cli --camera living_room --version-http-api
 version=1.40
 ```
+
+- TUI
+```bash
+Configure amcrest.conf and trigger amcrest-tui, make sure the user
+triggering amcrest-tui have access to framebuffer device or use sudo.
```
Comment (tchellomello) [inline README.md]:
```bash
↪ ./amcrest-tui 
Traceback (most recent call last):
  File "./amcrest-tui", line 188, in <module>
    with open(args.framebuffer_device, 'r') as f:
PermissionError: [Errno 13] Permission denied: '/dev/fb0'
```
I see that you added this on the documentation, but let's add the command to allow the user to have access to /dev/fb0

```bash
↪ ls -la /dev/fb0
crw-rw----. 1 root video 29, 0 Jan 31 01:03 /dev/fb0
↪ sudo usermod  -a -G video <user>
```

Comment (tchellomello) [inline README.md]:
```bash
↪ ./amcrest-tui 
Traceback (most recent call last):
  File "./amcrest-tui", line 188, in <module>
    with open(args.framebuffer_device, 'r') as f:
PermissionError: [Errno 13] Permission denied: '/dev/fb0'
```
I see that you added this on the documentation, but let's add the command to allow the user to have access to /dev/fb0

```bash
↪ ls -la /dev/fb0
crw-rw----. 1 root video 29, 0 Jan 31 01:03 /dev/fb0
↪ sudo usermod  -a -G video <user>
```

File: tui/amcrest-tui [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                            camera_name, 'password')).camera
+                }
+            except (NoSectionError, NoOptionError) as e:
+                print(("ERROR! %s found at %s" % (e, self.AMCREST_CONF)))
+                raise
+
+    def _view_image(self, image):
+        try:
+            cmd = subprocess.Popen(
+                ['fbi', '-a', '-noverbose', '-d', self.fb_device, image],
```
Comment (tchellomello) [inline tui/amcrest-tui]: What about to create a try/exception if the user does not have the `fbi` command? 

```bash
# ./amcrest-tui
Loading new image from: amcrestcam1
Traceback (most recent call last):
  File "./amcrest-tui", line 205, in <module>
    tui_viewer.run()
  File "./amcrest-tui", line 133, in run
    self._view_image(tmpfile.name)
  File "./amcrest-tui", line 99, in _view_image
    shell=False
  File "/usr/lib64/python3.5/subprocess.py", line 947, in __init__
    restore_signals, start_new_session)
  File "/usr/lib64/python3.5/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: 'fbi'
```

To address, I've installed fbi on my Fedora

```
↪ rpm -aq | grep fbida
fbida-2.09-10.fc24.x86_64
```

File: man/amcrest-tui.1 [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+amcrest-tui
+.SH DESCRIPTION
+\fBamcrest-tui\fP is python3 only text user interface to visualize snapshots from Amcrest cameras.
+.br
+By default it reads the default ~/.config/amcrest.conf configuration file.
+.SH NOTES:
+NOTE1: As it uses framebuffer to visualize the snapshots, make sure your user has permission to manage /dev/fb0.
+.br
+NOTE2: If you use sudo command, make sure you put the amcrest.conf in /root/.config/amcrest.conf
+.P
```
Comment (tchellomello) [inline man/amcrest-tui.1]: Let's add this note:

This option uses frrame buffer, which  requires your to run it directly from the Linux TTY console. Executing it from a  X terminal will not work.

File: README.md (hunk 1 above)
Comment (dougsland) [inline README.md]: I would recommend in a different patch.

File: tui/amcrest-tui (hunk 2 above)
Comment (dougsland) [inline tui/amcrest-tui]: For the fbi, we need to add a requirement for pip installs. For the second request, let's see if it's possible :)  I will do also in different patch.

File: man/amcrest-tui.1 (hunk 3 above)
Comment (dougsland) [inline man/amcrest-tui.1]: Sure, I will also do in a different patch.

**Comment by dougsland** [comment]
I will address the comments in others patches. For now merging as it worked for Marcelo too.

**Pull request merged**